jib-core/src/main/java/com/google/cloud/tools/jib/docker/DockerClient.java

```diff
@@ -165,13 +170,23 @@ public String load(ImageTarball imageTarball) throws InterruptedException, IOExc
    *     href="https://docs.docker.com/engine/reference/commandline/save/">https://docs.docker.com/engine/reference/commandline/save</a>
    * @param imageReference the image to save
    * @param outputPath the destination path to save the output tarball
+   * @param writtenByteCountListener callback to call when bytes are saved
    * @throws InterruptedException if the 'docker save' process is interrupted
    * @throws IOException if creating the tarball fails
    */
-  public void save(ImageReference imageReference, Path outputPath)
+  public void save(
+      ImageReference imageReference, Path outputPath, Consumer<Long> writtenByteCountListener)
       throws InterruptedException, IOException {
-    // Runs 'docker save'.
-    Process dockerProcess = docker("save", imageReference.toString(), "-o", outputPath.toString());
+    Process dockerProcess = docker("save", imageReference.toString());
+    try (InputStream stdout = new BufferedInputStream(dockerProcess.getInputStream());
+        OutputStream fileStream = new BufferedOutputStream(Files.newOutputStream(outputPath))) {
```

Comment:
Now that you used `NotifyingOutputStream` above (I completely forgot about it), can we do

```java
    OutputStream fileStream = new NotifyingOutputStream(..., writtenByteCountListener)) {
  ByteStreams.copy(stdout, fileStream);
}
```
? (Not tested.)